core/src/main/java/tachyon/UnderFileSystemSingleLocal.java

```diff
@@ -36,8 +36,15 @@
  */
 public class UnderFileSystemSingleLocal extends UnderFileSystem {
 
+  private final String mLocalhost;
+
   protected UnderFileSystemSingleLocal(TachyonConf tachyonConf) {
     super(tachyonConf);
+    
+    // Cache local host address to avoid having to determine it every time we want file locations
+    mLocalhost =
+        NetworkUtils.getLocalHostName(tachyonConf.getInt(Constants.HOST_RESOLUTION_TIMEOUT,
+            Constants.DEFAULT_HOST_RESOLUTION_TIMEOUT));
```

Comment:
This is mostly to avoid having to repeatedly look stuff up in the configuration, previously this was called in every `getFileLocations()` call.  Even with the caching in `NetworkUtils` since this would get called regularly I'd prefer to still cache it in this class